src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/armor/HazmatArmorPiece.java

```diff
@@ -0,0 +1,31 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.armor;
+
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.potion.PotionEffect;
+
+import io.github.thebusybiscuit.slimefun4.core.attributes.CustomProtection;
+import io.github.thebusybiscuit.slimefun4.core.attributes.ProtectionType;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+public class HazmatArmorPiece extends SlimefunArmorPiece implements CustomProtection {
+
+    private final ProtectionType[] types;
+
+    public HazmatArmorPiece(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe, PotionEffect[] effects, String setID) {
+        super(category, item, recipeType, recipe, effects, setID);
```

Comment:
Is the setID here really required? shouldn't it be `slimefun:hazmat` by default for this class?

Reply:
Thought an addon may extend this but mkay changing it.